### PR TITLE
Add zoom_user_users data source

### LINKED ISSUE #20: zoom_users data source
https://developers.zoom.us/docs/api/rest/reference/user/methods/#operation/users

### PR DESCRIPTION
**Pull request by kenchan0130**
Fix #20

### CODE REVIEW

File: internal/services/user/user/users_data_source.go [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+		MarkdownDescription: `Gets basic information for multiple Zoom users.
+
+## API Permissions
+
+The following API permissions are required in order to use this resource.
+This resource requires the ` + strings.Join([]string{
+			"`user:read:list_users:admin`",
+		}, ", ") + ".",
+		Attributes: map[string]schema.Attribute{
+			"status": schema.StringAttribute{
```
Comment (krrrr38) [inline internal/services/user/user/users_data_source.go:64]: how about to add `filter` SingleNestedAttribute to separate filter attribute and output attribute?

https://github.com/folio-sec/terraform-provider-zoom/blob/main/internal/services/phone/phonenumbers/phone_numbers_data_source.go#L60

```golang
  "filter": schema.SingleNestedAttribute {
    "status": ...
    "role_id": ...
  }
  "users: ...
```

File: internal/util/typesuser.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+func ToUserOptInt(o types.Int32) zoomuser.OptInt {
+	if o.IsNull() || o.IsUnknown() {
+		return zoomuser.OptInt{}
+	}
+	return zoomuser.NewOptInt(int(o.ValueInt32()))
+}
+
+func ToUserOptDateTime(o timetypes.RFC3339) zoomuser.OptDateTime {
```
Comment (krrrr38) [inline internal/util/typesuser.go:37]: > timetype

nice

File: internal/services/user/user/users_data_source.go (hunk 1 above)
Comment (kenchan0130) [reply internal/services/user/user/users_data_source.go:64]: I have no objection to nesting.
On the other hand, the parameter `include_fields` is closer in meaning to **select** than to **filter**.
Therefore, the safe name `query` may be acceptable.

What do you think this?

Comment (krrrr38) [reply internal/services/user/user/users_data_source.go:64]: `include_fields`'s use-case is interesting. We should make it flexisible on data source. Let's migrate it using `query`.

Comment (kenchan0130) [reply internal/services/user/user/users_data_source.go]: I have fixed the attribute. pls check it.

**Review by krrrr38 — APPROVED** [review]
lgtm

**Pull request merged**